### ISSUE TITLE
Non-X11 images

### ISSUE THREAD

**Issue by jwinarske** [OE4T/tegra-demo-distro#186]
I'm building Weston image without X11.  Nano and NX.

CI jobs:
https://github.com/meta-flutter/meta-flutter/blob/jw/nvidia_nx/.github/workflows/dunfell-nvidia-jetson-nano-devkit.yml
https://github.com/meta-flutter/meta-flutter/blob/jw/nvidia_nx/.github/workflows/dunfell-nvidia-jetson-xavier-nx-devkit.yml

X11 appears to be tightly coupled with demo-image-weston, and the build is ignoring this:
echo 'DISTRO_FEATURES_remove = "x11"' >> ./conf/local.conf

If DISTRO_FEATURES doesn't have `x11`, it shouldn't build anything with X11 dependency.

Same with vulkan recipes, they're all dependent on x11; where the actual repo supports backend of direct2display, xcb, x11, and Wayland.

Basically adding DISTRO_FEATURES_remove = "x11" should no longer include any x11 dependent recipes in the image.

What's the by-design approach for non-X11 images?

**Comment by jwinarske** [comment]
To get vulkaninfo and Sascha Willems samples (latest configured for Wayland) running on Wayland I needed to make this change

From
```
root@jetson-xavier-nx-devkit:/# cat /etc/vulkan/icd.d/nvidia_icd.json
{
    "file_format_version" : "1.0.0",
    "ICD": {
        "library_path": "libGLX_nvidia.so.0",
        "api_version" : "1.2.131"
    }
}
```

To
```
root@jetson-xavier-nx-devkit:/# cat /etc/vulkan/icd.d/nvidia_icd.json
{
    "file_format_version" : "1.0.0",
    "ICD": {
        "library_path": "/usr/lib/libnvvulkan-producer.so",
        "api_version" : "1.2.131"
    }
}
```

**Comment by madisongh** [comment]
Interesting, thanks.  The `nvidia_icd.json` file is copied out of the L4T package, so the GLX library is the one that NVIDIA specifies there as the provider.  If directly specifying the other library works, great.

There are several dependencies on x11 in L4T, and they're not all easy to tease out. Since NV's targeting Ubuntu with their prebuilt binaries, x11 is always present.

**Comment by jwinarske** [comment]
@madisongh I understand.  It would be good to have an active list of items that depend on X11.  I would expect NVidia to eventually move away from X11; perhaps they have no plans.  Ubuntu 20 Gnome window manager defaults to Wayland, and X11 is emulated (XWayland).  Between the two you get higher framerates with Wayland.

**Comment by Arech** [comment]
Gentlemen, excuse me for an offtopic, I'm looking for solution to another problem, but relevant resources are extremely scarse, so trying every opportunity. Do you know how to get rid of `nvdc` spamming these `nvdc: start nvdcEventThread` and `nvdc: exit nvdcEventThread` on Vulkan instance creation/destruction?

Also could that mean that Vulkan drivers are actually in a debug mode (and not production quality performance optimized)?

**Comment by ichergui** [comment]
Hey @jwinarske 

Any updates regarding this issue ? Otherwise, I will close it

**Comment by jwinarske** [comment]
@ichergui No update, other than I haven't tried a recent release.

**Comment by ichergui** [comment]
I think we should close this issue and open new one if needed when trying with new `L4T` release
